adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractVertxBasedCoapAdapter.java

```diff
@@ -155,6 +167,23 @@ public final void doStart(final Promise<Void> startPromise) {
         Optional.ofNullable(server)
                 .map(Future::succeededFuture)
                 .orElseGet(this::createServer)
+                .onSuccess(serverToStart -> serverToStart.setMessageDeliverer(
+                        new ServerMessageDeliverer(serverToStart.getRoot()) {
+                            @Override
+                            protected boolean preDeliverRequest(final Exchange exchange) {
+                                final Resource resource = findResource(exchange);
+                                if (resource != null) {
+                                    runOnContext(s -> resource.handleRequest(exchange));
```

Comment:
With Californium 3.x this may be also done by using the Resource's Executor, e.g. in [adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/TracingSupportingHonoResource.java](https://github.com/bosch-io/hono/commit/4e786177808bf7d328c0ced387ae0da39f125d04#diff-93a9c5ee588b4a66667390d15798721aea8ec0506e17d53fa5da4308f4822902) 

```
...
this.executor = new Executor() {
            @Override
            public void execute(final Runnable command) {
                adapter.runOnContext(s -> command.run());
            }
        };

...
@Override
public Executor getExecutor() {
      return executor;
}
```